flytekit/common/tasks/sagemaker/training_job_task.py

```diff
@@ -0,0 +1,199 @@
+
+    @property
+    def training_job_model(self) -> _training_job_models.TrainingJob:
+        return self._training_job_model
+
+
+class SdkCustomTrainingJobTask(_sdk_runnable.SdkRunnableTask):
+    def __init__(
+            self,
+            task_function: Callable,
+            training_job_resource_config: _training_job_models.TrainingJobResourceConfig,
+            algorithm_specification: _training_job_models.AlgorithmSpecification,
+            cache_version: str,
+            retries: int = 0,
+            # interruptible: bool = False,
+            deprecated: bool = False,
+            cacheable: bool = False,
+            # environment: Dict[str, str] = None,
+    ):
+        """
+
+        :param task_function:
+
+        :param training_job_resource_config: The options to configure the training job
+        :param algorithm_specification: The options to configure the target algorithm of the training
+        :param cache_version: String describing the caching version for task discovery purposes
+        :param retries: Number of retries to attempt
+        :param deprecated: This string can be used to mark the task as deprecated.  Consumers of the task will
+            receive deprecation warnings.
+        :param cacheable: The flag to set if the user wants the output of the task execution to be cached
+        """
+        # Use the training job model as a measure of type checking
+        training_job = _training_job_models.TrainingJob(
+            algorithm_specification=algorithm_specification,
+            training_job_resource_config=training_job_resource_config,
+        ).to_flyte_idl()
+
+        # Setting flyte-level timeout to 0, and let SageMaker takes the StoppingCondition and terminate the training
+        # job gracefully
+        timeout = _datetime.timedelta(seconds=0)
+
+        super(SdkCustomTrainingJobTask, self).__init__(
+            task_function=task_function,
+            task_type=SdkTaskType.SAGEMAKER_TRAINING_JOB_TASK,
+            discovery_version=cache_version,
+            retries=retries,
+            interruptible=False,
+            deprecated=deprecated,
+            storage_request="",
+            cpu_request="",
+            gpu_request="",
+            memory_request="",
+            storage_limit="",
+            cpu_limit="",
+            gpu_limit="",
+            memory_limit="",
+            discoverable=cacheable,
+            timeout=timeout,
+            environment={},
+            custom=MessageToDict(training_job),
+        ),
+        self.add_inputs(
+            {
+                "static_hyperparameters": _interface_model.Variable(
+                    type=_idl_types.LiteralType(simple=_idl_types.SimpleType.STRUCT),
+                    description="",
+                ),
+                "train": _interface_model.Variable(
+                    type=_idl_types.LiteralType(
+                        blob=_core_types.BlobType(
+                            format="csv",
```

Comment:
same thing, I think we should take the "data-format" and default it to csv. We should provide the options (enum)? Currently in the Enum you can have only csv